docs/DesignDocs/extra_enclave_data.md

```diff
@@ -0,0 +1,142 @@
+Extra SGX enclave data
```

Comment:
since this is SGX specific, the .md filename should also be updated to have sgx in it